File: cs/competitive_programming/leetcode_fulltimes/2023/januarary/926_flip_string_to_monotone_increasing.py
class Solution:
    def minFlipsMonoIncr(self, s: str) -> int:

        flips, ones = 0, 0

        for c in s:
            if c=='0': 
                flips+=1
            else:
                ones+=1
            flips = min(flips, ones)
        return flips

File: cs/competitive_programming/leetcode_fulltimes/2023/januarary/cpp/926_flip_string_to_monotone_increasing.cpp
#include <stdio.h>
using namespace std;
#include <string>

class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int flips = 0;
        int ones = 0;

        for(auto x: s){
            if(x=='0'){
                flips += 1;
            }
            else{
                ones += 1;
            }

            flips = min(flips, ones);
        }
        return flips;
    }
};